apis/v1alpha1/snippetsfilter_types.go

```diff
@@ -0,0 +1,110 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:categories=nginx-gateway-fabric,shortName=snippetsfilter
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// SnippetsFilter is a filter that allows inserting NGINX configuration into the
+// generated NGINX config for HTTPRoute and GRPCRoute resources.
+type SnippetsFilter struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the SnippetsFilter.
+	Spec SnippetsFilterSpec `json:"spec"`
+
+	// Status defines the state of the SnippetsFilter.
+	Status SnippetsFilterStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// SnippetsFilterList contains a list of SnippetFilters.
+type SnippetsFilterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []SnippetsFilter `json:"items"`
+}
+
+// SnippetsFilterSpec defines the desired state of the SnippetsFilter.
+type SnippetsFilterSpec struct {
+	// Snippets is a list of NGINX configuration snippets.
+	// There can only be one snippet per context.
+	// Allowed contexts: main, http, http.server, http.server.location.
+	Snippets []Snippet `json:"snippets"`
+}
+
+// Snippet represents an NGINX configuration snippet.
+type Snippet struct {
+	// Context is the NGINX context to insert the snippet into.
+	Context NginxContext `json:"context"`
+
+	// Value is the NGINX configuration snippet.
+	Value string `json:"value"`
+}
+
+// SnippetsFilterStatus defines the state of SnippetsFilter.
+type SnippetsFilterStatus struct {
+	// Conditions describes the state of the SnippetsFilter.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=8
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// SnippetsFilterConditionType is a type of condition associated with SnippetsFilter.
```

Comment:
nit: can we move the condition stuff below the context stuff since spec comes  before status in the struct?

For example: https://github.com/nginxinc/nginx-gateway-fabric/blob/main/apis/v1alpha1/nginxgateway_types.go